src/diet.ts

```diff
@@ -147,140 +185,49 @@ function getBestSpleenItems() {
 
 function fillSomeSpleen() {
   const { bestSpleenItem } = getBestSpleenItems();
-  print(`Spleen item: ${bestSpleenItem}`);
   fillSpleenWith(bestSpleenItem);
 }
 
 function fillSpleenWith(spleenItem: Item) {
-  if (mySpleenUse() + spleenItem.spleen <= spleenLimit()) {
-    // (adventureGain * spleenA + adventures) * 1.04 + 40 = 30 * spleenB + synthTurns
-    // spleenA + spleenB = spleenTotal
-    // (adventureGain * (spleenTotal - spleenB) + adventures) * 1.04 + 40 = 30 * spleenB + synthTurns
-    // 1.04 * adventureGain * (spleenTotal - spleenB) + 1.04 * adventures + 40 = 30 * spleenB + synthTurns
-    // 1.04 * adventureGain * spleenTotal - 1.04 * adventureGain * spleenB + 1.04 * adventures + 40 = 30 * spleenB + synthTurns
-    // 1.04 * adventureGain * spleenTotal + 1.04 * adventures + 40 = 30 * spleenB + synthTurns + 1.04 * adventureGain * spleenB
-    // (1.04 * adventureGain * spleenTotal + 1.04 * adventures + 40 - synthTurns) / (30 + 1.04 * adventureGain) = spleenB
+  if (mySpleenUse() < spleenLimit()) {
+    // (itemAdvs * spleenItem + adventures) * 1.04 = 30 * spleenSynth + synthTurns
+    // spleenItem + spleenSynth = spleenTotal
+    // (itemAdvs * (spleenTotal - spleenSynth) + adventures) * 1.04 = 30 * spleenSynth + synthTurns
+    // 1.04 * itemAdvs * (spleenTotal - spleenSynth) + 1.04 * adventures = 30 * spleenSynth + synthTurns
+    // 1.04 * itemAdvs * spleenTotal - 1.04 * itemAdvs * spleenSynth + 1.04 * adventures = 30 * spleenSynth + synthTurns
+    // 1.04 * itemAdvs * spleenTotal + 1.04 * adventures = 30 * spleenSynth + synthTurns + 1.04 * itemAdvs * spleenSynth
+    // (1.04 * itemAdvs * spleenTotal + 1.04 * adventures - synthTurns) / (30 + 1.04 * itemAdvs) = spleenSynth
     const synthTurns = haveEffect($effect`Synthesis: Greed`);
     const spleenTotal = spleenLimit() - mySpleenUse();
-    const adventuresPerItem = adventureGain(spleenItem);
+    const adventuresPerSpleen = getAverageAdventures(spleenItem) / spleenItem.spleen;
     // when not barfing, only get synth for estimatedTurns() turns (ignore adv gain)
-    const spleenAdvsGained = globalOptions.noBarf ? 0 : 1.04 * adventuresPerItem * spleenTotal;
+    const spleenAdvsGained = globalOptions.noBarf ? 0 : 1.04 * adventuresPerSpleen * spleenTotal;
     const spleenSynth = Math.ceil(
-      (spleenAdvsGained + estimatedTurns() - synthTurns) / (30 + 1.04 * adventuresPerItem)
+      (spleenAdvsGained + estimatedTurns() - synthTurns) / (30 + 1.04 * adventuresPerSpleen)
```

Comment:
if we eat a mix of sliders and nonsliders (or drink a combination of pickles and nonpickles), would it be better to use estimatedTurnsWithOrgans?

Reply:
yep!